shared/reducers/route-tree.js

```diff
@@ -110,7 +110,7 @@ function routeStateReducer(routeDef, routeState, action) {
     }
 
     case Constants.refreshRouteDef: {
-      return routeNavigate(action.payload.routeDef, routeState, getPath(routeState))
+      return routeNavigate(action.payload.appRouteTree, routeState, getPath(routeState))
```

Comment:
Ah, looks like this was a bug in the old code. We don't always want the first arg to be `appRouteTree`, though -- I think you want to look at the `refreshRouteDef` case in `routeDefReducer` and factor out that logic to pick the right `routeDef`, and then call that here.